packages/core/src/dynamoDb/utils/editItem.ts

```diff
@@ -0,0 +1,158 @@
+import { telemetry } from '../../shared/telemetry'
+
+export async function editItem(
+    selectedRow: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema }
+) {
+    const currentContent = JSON.stringify(selectedRow, undefined, 4)
+    let filename = `${selectedRow[tableSchema.partitionKey.name]}`
+    if (tableSchema.sortKey) {
+        filename += `-${selectedRow[tableSchema.sortKey.name]}`
+    }
+    const tempFilePath = path.join(os.tmpdir(), `${filename}.json`)
+    fs.writeFileSync(tempFilePath, currentContent)
+
+    const document = await vscode.workspace.openTextDocument(tempFilePath)
+    await vscode.window.showTextDocument(document, { preview: false, viewColumn: vscode.ViewColumn.Beside })
+
+    const saveListener = vscode.workspace.onDidSaveTextDocument(async (savedDoc) => {
+        if (savedDoc.fileName === tempFilePath) {
+            await saveItem(tempFilePath, selectedRow, { tableName, regionCode, tableSchema })
+        }
+    })
+
+    const closeListener = vscode.workspace.onDidCloseTextDocument(async (closedDoc) => {
+        if (closedDoc.fileName === tempFilePath) {
+            await saveItem(tempFilePath, selectedRow, { tableName, regionCode, tableSchema })
+            saveListener.dispose()
+            closeListener.dispose()
+            if (fs.existsSync(tempFilePath)) {
+                fs.unlinkSync(tempFilePath)
+            }
+        }
+    })
+}
+
+async function saveItem(
+    tempFilePath: string,
+    selectedRow: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema }
+) {
+    try {
+        const updatedContent = JSON.parse(fs.readFileSync(tempFilePath, 'utf-8'))
+        const response = await updateDynamoDbItem(updatedContent, selectedRow, { tableName, regionCode, tableSchema })
+        if (response) {
+            void vscode.window.showInformationMessage('Item updated successfully in DynamoDB')
+        }
+    } catch (error) {
+        void vscode.window.showErrorMessage(`Failed to update item in DynamoDB: ${(error as any).message}`)
+    }
+}
+async function updateDynamoDbItem(
+    updatedItem: RowData,
+    currentItem: RowData,
+    { tableName, regionCode, tableSchema }: { tableName: string; regionCode: string; tableSchema: TableSchema },
+    client: DynamoDbClient = new DynamoDbClient(regionCode)
+) {
+    const partitionKey = tableSchema.partitionKey.name
+    const sortKey = tableSchema.sortKey?.name
+
+    if (updatedItem[partitionKey] !== currentItem[partitionKey]) {
+        void vscode.window.showErrorMessage(`Partition key '${partitionKey}' cannot be updated.`)
+        return
+    }
+
+    if (sortKey && updatedItem[sortKey] !== currentItem[sortKey]) {
+        void vscode.window.showErrorMessage(`Sort key '${sortKey}' cannot be updated.`)
+        return
+    }
+
+    if (!compareRowDataItems(updatedItem, currentItem)) {
```

Comment:
We might be able to throw an error here to tell the users that you can't add new keys in